### PR TITLE
Began rewrite of the BTP wireshark plugin in Lua

### PR DESCRIPTION
**Pull request by verifiedtm**
This rewrites the entire old BTP Wireshark plugin in Lua instead of C, makes appropriate changes to the documentation, and removes old build scripts that are no longer used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,38 +2,21 @@
...
-## Requirements
+## Installation
 
-- CMake >= 3.5 (>= 3.7 on 32 bit Windows)
-- Python >= 3.4
-- Qt >= 5
+To install the BTP Wireshark plugin, copy `btp.lua` to the correct path for your operation system
 
+ - Windows users can copy the file to `%APPDATA%\Wireshark\plugins\` or `WIRESHARK\plugins\`, where WIRESHARK is their Wireshark installation location.
+ - Unix-like users can copy the file to `~/.local/lib/wireshark/plugins`
```
Comment (wraithm) [inline README.md]: You might wanna just say "macOS, Linux, and other unix-like" or something like that.

File: btp.lua [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+-- This function matches packets against a series of tests to determine whether
+-- or not the packet is a BTP packet
+local function heuristic_checker(buffer, pinfo, tree)
+    -- ensure that we have at least the minimum packet size
+    length = buffer:len()
+    if length < 12 then return false end
+
+    -- ensure that the protocol ID matches
+    protocol = buffer:range(0, 2):string()
+    if protocol ~= "BT" then return false end
```
Comment (wraithm) [inline btp.lua:474]: Should we check the protocol version too?

File: README.md (hunk 1 above)
Comment (verifiedtm) [reply README.md]: I clarified instructions for macOS and Linux, and fixed the broken link to the wireshark manual

File: btp.lua (hunk 2 above)
Comment (verifiedtm) [reply btp.lua:474]: That sounds right to me, I added the check

File: btp.lua [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+                                                 "Message Type", base.HEX,
+                                                 price_feed_message_type_string,
+                                                 nil, nil)
+local login_message_type = ProtoField.uint8("btp.login.message_type",
+                                            "Message Type", base.HEX,
+                                            login_message_type_string, nil, nil)
+
+local function dissect_order_entry(buffer, pinfo, tree)
+    tree:add_le(order_entry_message_type, buffer:range(0, 1))
+    local mt = buffer:range(0, 1):string()
```
Comment (wraithm) [inline btp.lua]: You get `buffer:range(0, 1)` twice, might as well put that in a variable, like `mt_byte`?

File: btp.lua [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+}
+
+local drop_copy_message_type_string = {
+    [string.byte("O")] = "Open Ack",
+    [string.byte("M")] = "Modify Ack",
+    [string.byte("F")] = "Fill",
+    [string.byte("C")] = "Close"
+}
+
+local price_feed_message_type_string = {
```
Comment (wraithm) [inline btp.lua:84]: nit: We've typically had `pricefeed` as one word in our system. Maybe a little strange, but that's what we do.

Comment (wraithm) [inline btp.lua]: nit: We typically have `pricefeed` as one word.

File: btp.lua [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+local order_entry_message_type = ProtoField.uint8(
+                                     "btp.order_entry.message_type",
+                                     "Message Type", base.HEX,
+                                     order_entry_message_type_string, nil, nil)
+local drop_copy_message_type = ProtoField.uint8("btp.drop_copy.message_type",
+                                                "Message Type", base.HEX,
+                                                drop_copy_message_type_string,
+                                                nil, nil)
+
+local price_feed_message_type = ProtoField.uint8("btp.price_feed.message_type",
```
Comment (wraithm) [inline btp.lua]: nit: typically we have `pricefeed` as one word.

File: btp.lua (hunk 3 above)
Comment (verifiedtm) [reply btp.lua]: Yep, fixed

File: btp.lua (hunk 5 above)
Comment (verifiedtm) [reply btp.lua]: Fixed

File: btp.lua (hunk 4 above)
Comment (verifiedtm) [reply btp.lua]: Fixed

**Review by wraithm — APPROVED** [review]

**Pull request merged**